### PR TITLE
Filter out inconsistent V2 consents on forms

### PR DESCRIPTION
**Pull request by walaura**
## What does this change?
At the moment, when a user has consents for things that are not on the consent model the identity application will crash. This PR filters out any consents that are missing from the model before they can crash, and logs them so any errors related to them are retrievable.

## What is the value of this and can you measure success?
A crash on this area of the application is very hard to recover from the user side because any list that could allow them to set their consents back to a consistent state will fail to load because thy are having wonky ones in the first place

### CODE REVIEW

File: identity/app/form/PrivacyMapping.scala [hunk 1]
```diff
@@ -108,7 +110,34 @@ case class PrivacyFormData(
...
+    * Checks if a consents is on the library and logs if not
+    *
+    * @param consent Consent
+    * @return Boolean (false if consent does not exist)
+    */
+  def checkIfConsentExistsInModel(consent:Consent): Boolean = {
+    Try(Consent.wording(consent.id, consent.version)) match {
+      case Success(wording:ConsentWording) => true
+      case Failure(f) => {
+        LogMissingConsent(consent)
```
Comment (walaura) [inline identity/app/form/PrivacyMapping.scala]: not a fan of this side effect, not sure how to refactor it in a way that doesn't just bury it deeper though :(

File: identity/app/form/PrivacyMapping.scala [hunk 2]
```diff
@@ -120,5 +149,6 @@ object PrivacyFormData {
       receiveGnmMarketing = userDO.statusFields.receiveGnmMarketing,
       receive3rdPartyMarketing = userDO.statusFields.receive3rdPartyMarketing,
       allowThirdPartyProfiling = userDO.statusFields.allowThirdPartyProfiling,
-      consents = if (userDO.consents.isEmpty) defaultConsents else userDO.consents)
+      consents = if (userDO.consents.isEmpty) defaultConsents else userDO.consents.filter(checkIfConsentExistsInModel)
```
Comment (walaura) [inline identity/app/form/PrivacyMapping.scala]: @mario-galic if i'm reading this correctly, a user with any set consents will never read from `defaultConsents` again. What is ensuring they can get any potential new consents from `identity`?

**Review by mariogalic — APPROVED** [review]

File: identity/app/form/PrivacyMapping.scala [hunk 3]
```diff
@@ -108,17 +110,41 @@ case class PrivacyFormData(
...
+    * Filter out any invalid consents that are still lingering in Mongo DB.
+    *
+    * For example, if consent id is renamed, then some users might still have the old consent.
+    *
+    * @param userDO Identity User domain model from IDAPI defiend that might contain some old invalid consents
+    * @return list of valid consents
+    */
+  private def onlyValidConsents(userDO: User): List[Consent] = {
+    def consentExistsInModel(consent:Consent): Boolean =
+      Try(Consent.wording(consent.id, consent.version)).isSuccess
```
Comment (jfsoul) [inline identity/app/form/PrivacyMapping.scala]: it feels like `Consent.wording` should probably have been returning an Option

Comment (mario-galic) [reply identity/app/form/PrivacyMapping.scala]: We will address it in a separate PR.

**Pull request merged**